lib/features/support/runner.rb

```diff
@@ -69,8 +70,32 @@ def run_script(script_name, blocking: false, success_codes: [0])
       return run_command(script_path, blocking: blocking, success_codes: success_codes)
     end
 
+    # Returns the current interactive session, creating a new one if necessary
+    #
+    # @return [InteractiveCLI] The interactiveCLI instance
+    def get_interactive_session
+      @interactive_session ||= InteractiveCLI.new(environment)
+      attempts = 0
+      until @interactive_session.pid
+        raise "Shell session would not start within 3 seconds" if attempts > 10 # allows 3 seconds of opening
+        attempts += 1
+        sleep(0.3)
+      end
+      @interactive_session
```

Comment:
This works but I find it a little confusing as it runs through the `until` even though it's already started; I think something like this is clearer (but doesn't have to be in this PR)

```suggestion
      return @interactive_session unless @interactive_session.nil?

      interactive_session = InteractiveCLI.new(environment)
      attempts = 0
      until @interactive_session.pid
        raise "Shell session would not start within 3 seconds" if attempts > 10 # allows 3 seconds of opening
        attempts += 1
        sleep(0.3)
      end

      @interactive_session = interactive_session
```

i.e. if there's a session already then return it straight away, otherwise make a new one

Reply:
Thanks, I've taken a note to address this in a follow-on PR.